routes/dao_coin_exchange_with_fees.go

```diff
@@ -681,28 +681,32 @@ func (fes *APIServer) GetBaseCurrencyPriceEndpoint(ww http.ResponseWriter, req *
 
 	// Iterate through the bids "filling" orders until we hit the base currency
 	// quantity we're looking for.
-	baseCurrencyFilled := float64(0.0)
-	baseCurrencyToFill := requestData.BaseCurrencyQuantityToSell
-	quoteCurrencyReceived := float64(0.0)
+	baseCurrencyFilled := big.NewFloat(0.0)
+	baseCurrencyToFill := big.NewFloat(requestData.BaseCurrencyQuantityToSell)
+	quoteCurrencyReceived := big.NewFloat(0.0)
 	for _, bid := range simpleBidOrders {
 		// If the amount filled plus the amount we're about to fill is greater
 		// than the amount we're looking to fill, then we just partially fill
 		// the order.
-		if baseCurrencyFilled+bid.AmountInBaseCurrency > baseCurrencyToFill {
-			baseCurrencyFromThisOrder := baseCurrencyToFill - baseCurrencyFilled
-			quoteCurrencyReceived += baseCurrencyFromThisOrder * bid.PriceInQuoteCurrency
+		if big.NewFloat(0.0).Add(
+			baseCurrencyFilled,
+			big.NewFloat(bid.AmountInBaseCurrency),
+		).Cmp(baseCurrencyToFill) > 1 {
```

Comment:
Shouldn't this be >0 rather than >1? I can't remember but I think it returns -1 = "less than" and 1 = "greater than" so you want to Cmp to zero.